addons/account_payment/controllers/payment.py

```diff
@@ -19,6 +20,11 @@ def invoice_pay_form(self, acquirer_id, invoice_id, save_token=False, access_tok
         success_url = kwargs.get('success_url', '/my')
         callback_method = kwargs.get('callback_method', '')
 
+        success_url_dict = urls.url_decode(success_url.split('?')[1])
```

Comment:
I don't really understand what you do. If I understand well there are some parameters propagated from payment to payment when done multiple times. Should'tn we try to avoid this parameter propagation instead of doing that kind of hackish code here ? Why are the success and error urls badly formatted ?

Reply:
In fact the function that call this route simply propagate all the parameters. Instead of remove them in the xml file, I prefered to clean it here, so I'm sure to don't lose other parameters

Reply:
It makes the code difficult to understand. Where does to extra parameters come from ?

Reply:
it comes from here: https://github.com/odoo/odoo/pull/21971/files#diff-b9586b5244cdc1451fca18f637b896a7 (line 45 of new lines)